### PR TITLE
Remove resuming multipart uploads

### PR DESCRIPTION
**Pull request by harshavardhana**
ListParts is never fully complete in its output, there is always a
possibility that the List is partial where another
PutObjectPart() is being uploaded in parallel.

While it may seem like ListParts and ListMultipartUploads
provide the possibility of resuming an upload, it would
be a mistake as these APIs were inherently never meant do
these things. This is one of the reasons why Amazon
recommends that one should avoid using these API calls to
complete or add parts to an upload - they instead
recommend to remember the parts being uploaded at the client.

### CODE REVIEW

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -129,7 +129,7 @@
  * </ul>
  * <h2>Bucket</h2>
  * <ul>
- * <li> Creating an object, including automatic upload resuming for large objects.</li>
+ * <li> Creating an object, including automatic multipart for large objects.</li>
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: many build failures need to be fixed

Comment (harshavardhana) [reply api/src/main/java/io/minio/MinioClient.java]: Done fixed

Comment (balamurugana) [reply api/src/main/java/io/minio/MinioClient.java]: travis fails
```
Test: copyObject(String bucketName, String objectName, String destBucketName,CopyConditions copyConditions) with Matching ETag (Positive Case)
error occurred
ErrorResponse(code=PreconditionFailed, message=At least one of the pre-conditions you specified did not hold, bucketName=null, objectName=null, resource=/minio-java-test-qbhgnt/minio-java-test-c25qkq, requestId=1571B4E0D0E41A04, hostId=3L137)
request={method=PUT, url=https://play.minio.io:9000/minio-java-test-qbhgnt/minio-java-test-c25qkq, headers=x-amz-copy-source-if-match: 7716b449206546dc4f9b614eb73b9b0f
x-amz-copy-source: minio-java-test-79vchn/minio-java-test-c25qkq
Content-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==
Host: play.minio.io:9000
User-Agent: Minio (amd64; amd64) minio-java/dev
x-amz-content-sha256: UNSIGNED-PAYLOAD
x-amz-date: 20181219T102344Z
Authorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20181219/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*
}
response={code=412, headers=Accept-Ranges: bytes
Content-Security-Policy: block-all-mixed-content
Content-Type: application/xml
Etag: "20000f00c7040abe235141e077909f6a6cfd86b2060db68294ffa2a307729442a80f51f01dd8a81b1dfc47640f30b87f"
Last-Modified: Wed, 19 Dec 2018 10:23:44 GMT
Server: Minio/DEVELOPMENT.2018-12-17T04-36-21Z (linux; amd64)
Vary: Origin
X-Amz-Bucket-Region: us-east-1
X-Amz-Request-Id: 1571B4E0D0E41A04
X-Minio-Deployment-Id: c3725876-4560-4ded-aba3-d360b31e6e12
X-Xss-Protection: 1; mode=block
Date: Wed, 19 Dec 2018 10:23:44 GMT
Transfer-Encoding: chunked
}
	at io.minio.MinioClient.executeReq(MinioClient.java:1192)
	at io.minio.MinioClient.execute(MinioClient.java:1050)
	at io.minio.MinioClient.executePut(MinioClient.java:1391)
	at io.minio.MinioClient.executePut(MinioClient.java:1413)
	at io.minio.MinioClient.copyObject(MinioClient.java:2191)
	at io.minio.MinioClient.copyObject(MinioClient.java:2011)
	at FunctionalTest.copyObject_test3(FunctionalTest.java:2110)
	at FunctionalTest.runTests(FunctionalTest.java:2979)
	at FunctionalTest.main(FunctionalTest.java:3094)
```

**Comment by sinhaashish** [comment]
Tested and LGTM

**Pull request merged**